src/components/Label/Label.tsx

```diff
@@ -65,29 +68,22 @@ class Label extends UIComponent<any, any> {
     as: 'label',
   }
 
-  static rules = labelRules
+  handleIconOverrides = iconProps => {
+    const { onIconClick, iconPosition, content, variables } = this.props
 
-  static variables = labelVariables
-
-  handleIconOverrides = predefinedProps => {
-    const { onIconClick, iconPosition, content, variables: labelPropVariables } = this.props
-    const { onClick, variables, xSpacing } = predefinedProps
-
-    const iconVariables = callable(variables)()
-    const labelVariables = labelPropVariables
-      ? callable(labelPropVariables)()
-      : callable(Label.variables)()
+    const iconVariables = callable(iconProps.variables)() || {}
+    const labelVariables = callable(variables)() || {}
```

Comment:
This may also need careful review.

The iconProps available at this point are produced by the shorthand factory.  Any variables provided on iconProps at this point will not have access to the theme values on context so they cannot be resolved before being handed to the overrides function.

This means we will not be able to hand resolved variables to components reliably, unless we also make factories aware of how to resolve variables (which I think is a very bad idea).  That would require making factories access context.

/cc @kuzhelov @mnajdova

Reply:
yes, agree with increasing complexity to address merge conflicts - will help you with merging this PR today.